hudi-common/src/main/java/org/apache/hudi/common/table/read/UpdateProcessor.java

```diff
@@ -39,8 +52,16 @@ public interface UpdateProcessor<T> {
   BufferedRecord<T> processUpdate(String recordKey, BufferedRecord<T> previousRecord, BufferedRecord<T> mergedRecord, boolean isDelete);
 
   static <T> UpdateProcessor<T> create(HoodieReadStats readStats, HoodieReaderContext<T> readerContext,
-                                       boolean emitDeletes, Option<BaseFileUpdateCallback<T>> updateCallback) {
-    UpdateProcessor<T> handler = new StandardUpdateProcessor<>(readStats, readerContext, emitDeletes);
+                                       boolean emitDeletes, Option<BaseFileUpdateCallback<T>> updateCallback,
+                                       TypedProperties properties) {
+    UpdateProcessor<T> handler;
+    Option<String> payloadClass = readerContext.getPayloadClasses(properties).map(Pair::getRight);
```

Comment:
`HoodeReaderContext.getPayloadClasses()` will not return payload class unless the merge strategyId is = PAYLOAD_BASED_MERGE_STRATEGY_UUID right. 
so, once we land w/ table v9 table creation patch, and rebase this patch, we do not need to make any additional fixes in these lines?

Reply:
Yes, that is the idea. This is meant to cover the legacy payload flow only